File: 3493-maximum-number-of-operations-to-move-ones-to-the-end/3493-maximum-number-of-operations-to-move-ones-to-the-end.cpp
class Solution {
public:
    int maxOperations(string s) {
        int res = 0;
        int onesCount = 0;
        for(int i = 0; i < s.length(); i++)
        {
            if(s[i] == '1')
            {
                onesCount++;
            } 
            else
            {
                while(i + 1 < s.length() && s[i + 1] != '1')
                {
                    i++;
                }
                res += onesCount;
            }
        }
        return res;
    }
};

File: 3493-maximum-number-of-operations-to-move-ones-to-the-end/3493-maximum-number-of-operations-to-move-ones-to-the-end.py
class Solution:
    def maxOperations(self, s: str) -> int:
        res = 0
        onesCount = 0
        i = 0
        while i < len(s):
            if s[i] == '1':
                onesCount += 1
            else:
                while i + 1 < len(s) and s[i + 1] != '1':
                    i += 1
                res += onesCount    
            i += 1    
        return res